src/python/pants/backend/docker/goals/package_image.py

```diff
@@ -347,7 +350,7 @@ def get_build_options(
                 source=source,
                 error_cls=DockerImageOptionValueError,
             )
-            yield from target[field_type].options(format)
+            yield from target[field_type].options(format, global_build_hosts_options=global_build_hosts_options)  # type: ignore[attr-defined]
```

Comment:
I assume this is a mypy bug but couldn't find any workaround. It seems to occur when there are more than 2 subclasses added to `issubclass`- e.g. splitting into separate `if` statements for each subclass doesn't trigger it.

Reply:
interesting. we can go with this for now.. I'll have a look on my end, if I'll find anything I can follow up in a new PR in that case.